src/components/Admin/EditOpportunity.tsx

```diff
@@ -0,0 +1,158 @@
+    awaitRefetchQueries: true,
+  });
+
+  const { data: opportunityProfile } = useOpportunityProfileInfoQuery({
+    variables: { ecoverseId: ecoverseId, opportunityId: opportunityNameId },
+    skip: mode === FormMode.create,
+  });
+
+  const opportunity = opportunityProfile?.ecoverse?.opportunity;
+  const opportunityId = useMemo(() => opportunity?.id || '', [opportunity]);
+
+  const isLoading = isCreating || isUpdating;
+
+  const currentPaths = useMemo(() => [...paths, { name: opportunity?.displayName || 'new', real: false }], [
+    paths,
+    opportunity,
+  ]);
+  useUpdateNavigation({ currentPaths });
+
+  const onSubmit = async (values: ProfileFormValuesType) => {
+    const { name, nameID, background, impact, tagline, vision, who, references, visual, tagsets } = values;
+
+    const updatedRefs: UpdateReferenceInput[] = references.map<UpdateReferenceInput>(r => ({
+      ID: r.id,
+      description: r.description,
+      name: r.name,
+      uri: r.uri,
+    }));
+
+    const updateContext: UpdateContextInput = {
+      background: background,
+      impact: impact,
+      references: updatedRefs,
+      tagline: tagline,
+      vision: vision,
+      visual: visual,
+      who: who,
+    };
+    const createContext: CreateContextInput = {
+      background: background,
+      impact: impact,
+      references: references,
+      tagline: tagline,
+      vision: vision,
+      visual: visual,
+      who: who,
+    };
+
+    switch (mode) {
+      case FormMode.create:
+        createOpportunity({
+          variables: {
+            input: {
+              nameID: nameID,
+              context: createContext,
+              displayName: name,
+              challengeID: challengeNameId,
+              tags: tagsets.map(x => x.tags.join()),
+            },
+          },
+        });
+        break;
+      case FormMode.update:
+        updateOpportunity({
+          variables: {
+            input: {
+              nameID: nameID,
+              context: updateContext,
+              displayName: name,
+              ID: opportunityId,
+              tags: tagsets.map(x => x.tags.join()),
+            },
+          },
+        });
+        break;
+      default:
+        throw new Error(`Submit mode expected: (${mode}) found`);
+    }
+  };
+
+  let submitWired;
+  return (
+    <Container>
+      <Typography variant={'h2'} className={'mt-4 mb-4'}>
+        {title}
+      </Typography>
+      <ProfileForm
+        isEdit={mode === FormMode.update}
+        name={opportunity?.displayName}
+        nameID={opportunity?.nameID}
+        tagset={opportunity?.tagset}
+        context={opportunity?.context}
+        onSubmit={onSubmit}
+        wireSubmit={submit => (submitWired = submit)}
+      />
+      <div className={'d-flex mt-4 mb-4'}>
+        <Button disabled={isLoading} className={'ml-auto'} variant="primary" onClick={() => submitWired()}>
+          {isLoading ? <Loading text={'Processing'} /> : 'Save'}
+        </Button>
```

Comment:
If we like this variant of the Button, it can be extracted in custom button and reused on other pages for consistency.